common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java

```diff
@@ -80,6 +82,43 @@ public static Geometry getPixelAsCentroid(GridCoverage2D raster, int colX, int r
         return  polygon.getCentroid();
     }
 
+    public static List<PixelRecord> getPixelAsCentroids(GridCoverage2D rasterGeom, int band) throws TransformException, FactoryException {
+        RasterUtils.ensureBand(rasterGeom, band);
```

Comment:
99% of this code is identical to getPixelsAsPointss. Can we abstract the common logic and re-use by these functions?